graphium/fingerprinting/data.py

```diff
@@ -0,0 +1,216 @@
+        data_seed: int = 42,
+        num_workers: int = 0,
+        device: str = "cpu",
+        mol_cache_dir: str = "./expts/data/cache",
+        fps_cache_dir: str = "./expts/data/cache",
+    ):
+        super().__init__()
+
+        assert benchmark is not None or df_path is not None, "Either benchmark or df_path must be provided"
+
+        self.pretrained_models = pretrained_models
+        self.task = task
+        self.benchmark = benchmark
+        self.df_path = df_path
+        self.batch_size = batch_size
+        self.split_type = split_type
+        self.splits_path = splits_path
+        self.split_val = split_val
+        self.split_test = split_test
+        self.data_seed = data_seed
+        self.num_workers = num_workers
+        self.device = device
+        self.mol_cache_dir = mol_cache_dir
+        self.fps_cache_dir = fps_cache_dir
+        if benchmark is not None:
+            # Check if benchmark naming is already implied in config
+            if f"{benchmark}/{task}" not in mol_cache_dir:
+                self.mol_cache_dir = f"{mol_cache_dir}/{benchmark}/{task}"
+            if f"{benchmark}/{task}" not in fps_cache_dir:
+                self.fps_cache_dir = f"{fps_cache_dir}/{benchmark}/{task}"
+
+        self.train_dataset = None
+        self.valid_dataset = None
+        self.test_dataset = None
+
+        self.splits = []
+
+    def prepare_data(self) -> None:  
+        if self.fps_cache_dir is not None and os.path.exists(f"{self.fps_cache_dir}/fps.pt"):
+            self.smiles, self.labels, self.fps_dict = torch.load(f"{self.fps_cache_dir}/fps.pt").values()
+            self.splits = list(self.smiles.keys())
+        
+        else:
+            # Check which splits are needed
+            self.splits = []
+            add_all = self.benchmark is not None or self.splits_path is not None
+            if add_all or self.split_val + self.split_test < 1:
+                self.splits.append("train")
+            if add_all or self.split_val > 0:
+                self.splits.append("valid")
+            if add_all or self.split_test > 0:
+                self.splits.append("test")
+
+            self.data = {
+                split: {
+                    "smiles": [],
+                    "labels": [],
+                    "fps": {},
+                }
+                for split in self.splits
+            }
+
+            for model, layers in self.pretrained_models.items():
+                predictor = PredictorModule.load_pretrained_model(model, device=self.device)
+                predictor.featurization.pop("max_num_atoms", None)
+
+                # Featurization
+                if self.benchmark is None:
+                    assert self.df_path is not None, "df_path must be provided if not using an integrated benchmark"
+
+                    # Add a dummy task column (filled with NaN values) in case no such column is provided
+                    smiles_df = pd.read_csv(self.df_path)
```

Comment:
Too restrictive. Can't finetune on parquet or other? Use the `MultitaskDataModule.read_table`?

Reply:
Using `MultitaskDataModule.read_table` now as suggested (see [here](https://github.com/datamol-io/graphium/pull/529/commits/d9aa407d774934e170e8d8af7028155b676b5981#diff-c2ceb6f93c9d4be9d916f0840dbd40f0af5be3566a18d13fd2857f0c20b537e3)).